pkg/virt-handler/vm.go

```diff
@@ -687,6 +690,31 @@ func (d *VirtualMachineController) migrationSourceUpdateVMIStatus(origVMI *v1.Vi
 	return nil
 }
 
+type NetworkStatus struct {
+	Name      string   `yaml:"name"`
+	Ips       []string `yaml:"ips"`
+	Interface string   `yaml:"interface"`
+}
+
+// Using the downward API, look for dedicated migration network net1 and, if found, set migration IP to its IP
+func findMigrationIP(vmi *v1.VirtualMachineInstance, migrationIp *string) {
+	var networkStatus []NetworkStatus
+
+	dat, err := ioutil.ReadFile("/etc/podinfo/network-status")
+	if err != nil {
+		return
```

Comment:
Let's see that we return errors. I would suggest that you detect that we are supposed to use a multus network during the startup phase and only then call this function. Then if you see an error, fail the whole handler start.